lib/jormungandr/src/Cardano/Pool/Jormungandr/Metrics.hs

```diff
@@ -139,6 +145,48 @@ data Block = Block
     -- ^ Any stake pools that were registered in this block.
     } deriving (Eq, Show, Generic)
 
+
+-- | @StakePoolLayer@ is a thin layer ontop of the DB. It is /one/ value that
+-- can easily be passed to the API-server, where it can be used in a simple way.
+data StakePoolLayer e m = StakePoolLayer
```

Comment:
don't we want to have StakePoolLayer interface in `core`? And in jormunagdr and shelley implementations of it?

Reply:
I believe we/you currently only need the `knownStakePools :: IO [PoolId]` field to check that the pool we're joining actually exists. So I see no point in keeping the entire `StakePoolLayer` then, since it would just be more work trying to unify the different pool-types under one common interface.

Reply:
a ok, got it . so now we will sneak `IO [PoolId]` instead of `StakePoolLayer` on wallet level. and just call it (with liftIO)

Reply:
Yeah (see diff at https://github.com/input-output-hk/cardano-wallet/pull/1738#discussion_r437978274)

I was curious about using a `IO (PoolId -> Bool)` also, but didn't do it.